src/ImageSharp/Processing/Processors/Convolution/Convolution2DProcessor.cs

```diff
@@ -56,17 +57,19 @@ protected override void OnApply(ImageBase<TPixel> source, Rectangle sourceRectan
             int maxY = endY - 1;
             int maxX = endX - 1;
 
-            using (PixelAccessor<TPixel> targetPixels = new PixelAccessor<TPixel>(source.Width, source.Height))
-            using (PixelAccessor<TPixel> sourcePixels = source.Lock())
+            using (var targetPixels = new PixelAccessor<TPixel>(source.Width, source.Height))
```

Comment:
We can use `Buffer2D` on these places, possibly replacing all usages of `PixelAccessor`. 
But this change might be big enough to keep it for another PR ..